manimlib/mobject/numbers.py

```diff
@@ -85,10 +85,15 @@ def init_data(self):
     def get_font_size(self):
         return self.data["font_size"][0]
 
-    def string_to_mob(self, string, mob_class=Text):
-        if string not in string_to_mob_map:
-            string_to_mob_map[string] = mob_class(string, font_size=1)
-        mob = string_to_mob_map[string].copy()
+    def string_to_mob(self, string, mob_class=Text, **kwargs):
+        if not kwargs:
+            if string not in string_to_mob_map:
+              string_to_mob_map[string] = mob_class(string, font_size=1, **kwargs)
+            mob = string_to_mob_map[string].copy()
+        else:
+            if (string, str(kwargs)) not in string_to_mob_map:
+              string_to_mob_map[(string, str(kwargs))] = mob_class(string, font_size=1, **kwargs)
+            mob = string_to_mob_map[(string, str(kwargs))].copy()
```

Comment:
Use `str` to make a dict hashable is not a good way.
And why not combine these two situations together?

Reply:
I was thinking not to combine because I am afraid to influence the speed for users who do not need to pass in  text_kwargs.
I changed the hashing way according to https://stackoverflow.com/questions/5884066/hashing-a-dictionary, how about now